addons/mail/static/src/core/web/chatter.js

```diff
@@ -253,22 +237,46 @@ export class Chatter extends Component {
     }
 
     /**
-     * @param {number} threadId
+     * Fetch data for the thread according to the request list.
+     * @param {import("models").Thread} thread
      * @param {['activities'|'followers'|'attachments'|'messages'|'suggestedRecipients']} requestList
      */
-    load(
-        threadId = this.props.threadId,
-        requestList = ["followers", "attachments", "messages", "suggestedRecipients"]
-    ) {
-        const { threadModel } = this.props;
-        this.state.thread = this.threadService.getThread(threadModel, threadId);
-        if (!threadId) {
+    load(thread, requestList = ["followers", "attachments", "messages", "suggestedRecipients"]) {
+        if (!thread.id || !this.state.thread?.eq(thread)) {
             return;
         }
         if (this.props.hasActivities && !requestList.includes("activities")) {
             requestList.push("activities");
         }
-        this.threadService.fetchData(this.state.thread, requestList);
+        this.threadService.fetchData(thread, requestList);
+    }
+
+    /**
+     * Visually refresh the chatter.
+     * @param {string} threadModel
+     * @param {number} threadId
+     * @param {Object} webRecord
+     */
+    changeThread(threadModel, threadId, webRecord) {
+        this.state.thread = this.threadService.getThread(threadModel, threadId);
+        this.state.thread.name = webRecord?.data?.display_name || undefined;
+        this.attachmentUploader.thread = this.state.thread;
+        if (threadId === false) {
+            this.state.composerType = false;
+        }
+        if (this.onNextUpdate) {
+            if (
+                !this.onNextUpdate({
```

Comment:
Using a side effect method in a condition is really unclear.
I would suggest to make the method return nothing. They all make a condition on `nextProps.threadId` before resetting, so it can be put here instead.

Reply:
`onNextUpdate` should really just be called `waitForThread`